# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1513312996 21600
#      Thu Dec 14 22:43:16 2017 -0600
# Branch stable
# Node ID 13344f00c564e2a76d903ec5097c08c363d1aeca
# Parent  11d7cd4ece5054e20cbe9eb82649b85f8342fbea
Protect being-deleted objects on figure list from second deletion (bug #52666)

* graphics.cc (gh_manager::do_delete_children): Check whether hchild is in
  the process of being deleted before freeing that child.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -1634,17 +1634,17 @@ children_property::do_get_children (bool
 
 void
 children_property::do_delete_children (bool clear)
 {
   for (auto& hchild : children_list)
     {
       graphics_object go = gh_manager::get_object (hchild);
 
-      if (go.valid_object ())
+      if (go.valid_object () && ! go.get_properties ().is_beingdeleted ())
         gh_manager::free (hchild);
     }
 
   if (clear)
     children_list.clear ();
 }
 
 bool
